docs/source/public_server.rst

```diff
@@ -23,56 +29,96 @@ Securing a notebook server
+    Verify password:
     Out[2]: 'sha1:67c9e60bb8b6:9ffede0825894254b2e042ea597d771089e11aed'
-    
-.. note::
 
-  :func:`~notebook.auth.security.passwd` can also take the password as a string
-  argument. **Do not** pass it as an argument inside an IPython session, as it
+.. caution::
+
+  :func:`~notebook.auth.security.passwd` when called with no arguments
+  will prompt you to enter and verify your password such as
+  in the above code snippet. Although the function can also
+  be passed a string as an argument such as ``passwd('mypassword')``, please
+  **do not** pass a string as an argument inside an IPython session, as it
   will be saved in your input history.
 
-You can then add this to your :file:`jupyter_notebook_config.py`, e.g.::
+Adding hashed password to your notebook configuration file
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+You can then add the hashed password to your :file:`jupyter_notebook_config.py`.
+The default location for this file :file:`jupyter_notebook_config.py` is in
+your Jupyter folder in your home directory, ``~/.jupyter``, e.g.::
 
-    # Password to use for web authentication
+    # Get notebook configuration and add hashed password
     c = get_config()
-    c.NotebookApp.password = 
+    c.NotebookApp.password =
     u'sha1:67c9e60bb8b6:9ffede0825894254b2e042ea597d771089e11aed'
 
-When using a password, it is a good idea to also use SSL, so that your 
-password is not sent unencrypted by your browser. You can start the notebook 
-to communicate via a secure protocol mode using a self-signed certificate with 
-the command::
+Using SSL for encrypted communication
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+When using a password, it is a good idea to also use SSL with a web certificate,
+so that your hashed password is not sent unencrypted by your browser.
 
-    $ ipython notebook --certfile=mycert.pem
+.. important::
+   Web security is rapidly changing and evolving. We provide this document
+   as a convenience to the user, and recommend that the user keep current on
+   changes that may impact security, such as new releases of OpenSSL.
+   The Open Web Application Security Project (`OWASP`_) website is a good resource
+   on general security issues and web practices.
 
-.. note::
+You can start the notebook to communicate via a secure protocol mode by setting
+the ``certfile`` option to your self-signed certificate, i.e. ``mycert.pem``,
+with the command::
+
+    $ jupyter notebook --certfile=mycert.pem
 
-    A self-signed certificate can be generated with ``openssl``.  For example, 
-    the following command will create a certificate valid for 365 days with 
+.. tip::
+
+    A self-signed certificate can be generated with ``openssl``.  For example,
+    the following command will create a certificate valid for 365 days with
     both the key and certificate data written to the same file::
 
         $ openssl req -x509 -nodes -days 365 -newkey rsa:1024 -keyout mycert.pem -out mycert.pem
```

Comment:
@rgbkrk @minrk I thought the `openssl` command should be `-keyout mycert.pem -out mycert.**csr**` can you actually output the key and the cert in the **same** file ?